### PR TITLE
Identity auth in discussion rendering

### PR DESCRIPTION
**Pull request by Georges-GNM**
Co-authored-by: @JamieB-gu 

## What does this change?

Following on from previous Okta/Identity-auth PRs, specifically the [latest refactor ](https://github.com/guardian/dotcom-rendering/pull/8119) and the [discussion-rendering into DCR move](https://github.com/guardian/dotcom-rendering/pull/8057), this enables various API calls in discussion-rendering to use Okta access/ID tokens where necessary when in the Okta experiment, as an alternative to the current method of the request being made with the `credentials: include` property. 

The API calls now take into account an `authStatus` parameter. This is a property of the `SignedInUser` type, which expands on the UserProfile type that exists in discussionTypes.ts. `SignedInUser` is now the type of the [`user` prop](https://github.com/guardian/dotcom-rendering/pull/8211/files#diff-0521f1dc80ca434dfbd843bbe156ffe6e0ba17272a9e9d27dc1b89281aedfb64R23) that gets passed down to discussion-rendering from the Discussion.tsx component in DCR. From there, the `authStatus` gets passed down to relevant components, e.g. [CommentForm.tsx,](https://github.com/guardian/dotcom-rendering/pull/8211/files#diff-e30f7d20b95c377cd00201c31c3f26e48c427a728325d0b187276292e6d18ec6R262) which then trigger the functions defined in [api.tsx](https://github.com/guardian/dotcom-rendering/pull/8211/files#diff-ae460a81df60e9adb882cddbba675359ef3ef98f869a762f3598024fd1fa55a4R137)

The functions which have been updated are:

- comment
- reply
- recommend
- pick and unpick
- reportAbuse
- addUserName

## Why?

This is the last piece for the Okta/DCR development work - next step is to migrate frontend.

### CODE REVIEW

**Review by georgeblahblah — APPROVED** [review]
This looks good to go once the [username change in identity API](https://github.com/guardian/identity/pull/2418) is merged.

One possible future change would be to leverage the type system to help model a signed out user for any discussion components where that's valid data.

For instance, in places where we have an optional `authStatus` prop[^1], we are saying we accept a signed in _or_ signed out user. A more specific type could help make that clearer, but I don't think it's necessary for this PR.

[^1]: https://github.com/guardian/dotcom-rendering/blob/gl-jb/identity-auth-in-discussion-rendering/dotcom-rendering/src/discussion-rendering/components/RecommendationCount/RecommendationCount.tsx#L58

**Comment by georgeblahblah** [comment]
Adding the "Do not merge" label until https://github.com/guardian/identity/pull/2418 is merged

**Comment by Georges-GNM** [comment]
The identity PR is now merged and so shall this one!

**Pull request merged**
